script.js

```diff
@@ -1,4 +1,215 @@
+//make deck of card
+//draw one card for player and dealer
+//player can analyze it and chose to hit or stand
+//if the player get over 21 the player lose
+var GAMEMODE = 'username'
+
+var playerTotal = 0
+var dealerTotal = 0
+var currentDeck
+var username
+var betNumber
+var playerPoint = 100
+var dealerPoint = 100
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if(GAMEMODE == 'username'){
+    GAMEMODE = 'input username'
+    return `input your username`
+  }
+  else if(GAMEMODE == 'input username'){
+    GAMEMODE = 'bet'
+    username = input
+    return `hello ${username}<br>you got 100 point, enter number of point you want to bet`
+  }
+   else if(GAMEMODE == 'bet'){
+    betNumber = Number(input)
+    if(playerPoint <= 0){
+      return 'you run out of point dealer winnn'
+    }
+    if(dealerPoint <= 0){
+      return 'dealer run out of point you winnn'
+    }
+    if(betNumber <= playerPoint && betNumber > 0){
+    GAMEMODE = 'draw card'
+    return wager(input)
+    }else{
+      return `please enter an appropriate number to bet<br>you got ${playerPoint} point`
+    }
+  }
+  else if(GAMEMODE == 'draw card'){
+    // reset the total
+    playerTotal = 0
+    dealerTotal = 0
+    // reset the deck
+    deck = shuffleDeck(makeDeck())
+
+    GAMEMODE = 'hit or stand'
+    return drawCardMode()
+  }
+  else if(GAMEMODE == 'ace'){
+    return aceChose(input)
+  }
+   else if(GAMEMODE == 'hit or stand'){
+    return hitOrStand(input)
+  }
+   else if(GAMEMODE == 'win or lose'){
+    return winOrLose()
+  }
 };
```

Comment:
this is a good use of modes in your game. Also, good naming of your game modes in a way that it is easy to comprehend